src/typing/typeloadModule.ml

```diff
@@ -192,8 +192,13 @@ end
 let module_pass_1 ctx m tdecls loadp =
 	let com = ctx.com in
 	let decls = ref [] in
-	let make_path name priv =
-		if List.exists (fun (t,_) -> snd (t_path t) = name) !decls then error ("Type name " ^ name ^ " is already defined in this module") loadp;
+	let make_path name priv p =
+		List.iter (fun (t2,(_,p2)) -> 
+			if snd (t_path t2) = name then begin
+				display_error ctx ("Type name " ^ name ^ " is already defined in this module") p;
```

Comment:
BTW we could still use `loadp` to report from where the type loading was initiated, but that's related to the discussions in #9052